### PR TITLE
Sign both test and prod assemblies

### PR DESCRIPTION
**Pull request by jskeet**
The private key is checked in, as recommended in
https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/strong-name-signing.md

The public key can be derived from the private one, so there's no need to check it in here.
The test assembly needs to be signed as the prod assembly exposes its internals to the test assembly.

### CODE REVIEW

File: src/Google.Api.Gax/project.json [hunk 1]
```diff
@@ -7,6 +7,9 @@
     "projectUrl": "htttps://github.com/googleapis/gax-dotnet",
     "licenseUrl": "https://developers.google.com/open-source/licenses/bsd",
     "tags": [ "Google" ],
+    "compilationOptions": {
+        "keyFile": "../../Gax.snk"
```
Comment (chrisdunelm) [inline src/Google.Api.Gax/project.json]: Unless I'm missing something, this ../../Gax.snk file doesn't seem to exist in this repo?

Comment (jskeet) [inline src/Google.Api.Gax/project.json]: It will when this PR is committed - it's the first file shown above, but with "binary file not shown" :)

Comment (chrisdunelm) [inline src/Google.Api.Gax/project.json]: Oh yes, my mistake :)

**Comment by chrisdunelm** [comment]
LGTM

**Pull request merged**